connector/src/main/java/org/jboss/as/connector/subsystems/datasources/GetInstalledDriverOperationHandler.java

```diff
@@ -59,18 +57,13 @@ public class GetInstalledDriverOperationHandler implements OperationStepHandler
 
     public static final GetInstalledDriverOperationHandler INSTANCE = new GetInstalledDriverOperationHandler();
 
-    private final ParametersValidator validator = new ParametersValidator();
-
     private GetInstalledDriverOperationHandler() {
-        validator.registerValidator(DRIVER_NAME.getName(), new StringLengthValidator(1));
     }
 
     @Override
     public void execute(final OperationContext context, final ModelNode operation) throws OperationFailedException {
 
-        validator.validate(operation);
```

Comment:
This should not be removed. A ParameterValidator should be used for input check, not ModelNode.require.

Reply:
@michpetrov Adding a validator to an AttributeDefinition doesn't cause it to be used. For it to be used, methods like AD.validateAndSet or AD.validateOperation need to be called. For the typical ADs that are used for resource attributes/add op params/write-attribute values, devs often don't see those calls because the base classes like AbstractAddStepHandler take care of that part.  But this is a custom OSH, so it needs to validate its inputs.

That said, I see now the old ParametersValidator thing was unnecessary. Changing L63 to this should handle it:

```
final String name = INSTALLED_DRIVER_NAME.validateOperation(operation).asString();
```